pkg/storage/storage.go

```diff
@@ -17,6 +17,14 @@ type ReadSeekCloser interface {
 	io.Closer
 }
 
+// Uploader upload file with chunks.
+type Uploader interface {
+	// UploadPart upload part of file data to storage
+	UploadPart(ctx context.Context, data []byte) error
+	// CompleteUpload make the upload data to a complete file
+	CompleteUpload(ctx context.Context) error
```

Comment:
IMO, to name this as `Close` to be consistent with other io like interface is 
a little better, anyway this is ok to me.